cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/media/class-video.php

```diff
@@ -338,19 +338,62 @@ public function print_video_scripts() {
 				}
 
 				$config = wp_parse_args( $video['args'], $default );
-				// Sizing.
-				if ( empty( $config['size'] ) ) {
+
+				if ( empty( $config['size'] ) && ! isset( $this->config['video_freeform'] ) ) {
 					$config['fluid'] = true;
 				}
-				$code[] = 'cld.videoPlayer(\'cloudinary-video-' . $instance . '\', ' . wp_json_encode( $config ) . ');';
+
+				$code[] = sprintf(
+					"var video%1\$s = cld.videoPlayer( \"cloudinary-video-%1\$s\", %2\$s );\n",
+					$instance,
+					wp_json_encode( $config )
+				);
```

Comment:
This does indeed fix the code repetition in the JS. I'll implement it this way. Thanks!

Reply:
@kienstra I tried implementing this, but it's a little troublesome as initializing the player onload is a little too late. Swapping the logic of the init outside the event causes weird behavior (only the last video instance receives the changes).

So I took your other suggestions for in this review for now, and I'll potentially revisit this in the future.

Reply:
Hi @dugajean,
Ah, good point.

For the script, how about something like:

```js

			ob_start();
			?>
			const cldVideos = <?php echo wp_json_encode( $cld_videos ); ?>;

			for ( const videoInstance in cldVideos ) {
				let config = cldVideos[ videoInstance ];
				let id = 'cloudinary-video-' + videoInstance;
				cld.videoPlayer( id, config );
			}

			window.addEventListener( 'load', function() {
				for ( const instance in cldVideos ) {
					let config = cldVideos[ instance ];
					let id = 'cloudinary-video-' + instance;
					let videoContainer = document.getElementById( id );
					let videoElement = videoContainer.getElementsByTagName( 'video' );

					if ( videoElement.length === 1 ) {
						videoElement = videoElement[0];
						videoElement.src = videoElement.src.replace( 'upload/', 'upload/' + instance + '/' );
					}
				}
			} );
			<?php
			$script = ob_get_clean();
```

This omits:

```js
if ( videoElement.width < 320 ) {
	video.controls(false);
}
```

You've looked at this more than me, so maybe that part is necessary.

But it looks like there already is a way to globally disable the controls:

![show-controls](https://user-images.githubusercontent.com/4063887/74054080-e09aea80-49a2-11ea-80f6-5cb800e4e4ea.png)

This seems like it should be handled by the user or the Cloudinary API, not this script.

But you've looked at this more than I have 😄

Reply:
Ah, I didn't explain that. I forcefully turnoff the controls because they go out of bounds when the player is less than 320px. This way we can ensure that we wont get a broken UI.

Reply:
I'll also test your code generation changes. Thanks for the input on that 😄

Reply:
OK, thanks for elaborating.

Reply:
Do you think this plugin should be disabling controls, when the user can presumably do that via that screenshot above? Or maybe here also:

<img width="694" alt="playback-controls" src="https://user-images.githubusercontent.com/4063887/74062797-b0f4de00-49b4-11ea-8a20-f84f19903a57.png">